src/controllers/Role.controller.ts

```diff
@@ -0,0 +1,132 @@
+      status: 200,
+      success: true,
+      data: allRolesAndPermissionsInDB,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while getting all roles",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// Admin set roles in db
+export const setRole = async (req: Request, res: Response) => {
+  try {
+    const role: RoleAndPermissionAttribute = req.body;
+
+    const settedRole = await setRoleServise(role);
+
+    return res.status(200).json({
+      status: 200,
+      success: true,
+      message: "Role setting is done successfully!",
+      data: settedRole,
+    });
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while seting role in db",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// Admin set role to a particular user
+export const setRoleToUser = async (req: Request, res: Response) => {
+  try {
+    const userId = req.params.id;
+
+    if (isValidUuid(req.body.roleId)) {
+      const updatedUserRole = await User.update(
+        { roleId: req.body.roleId },
+        {
+          where: { id: userId },
+          returning: true,
+        },
+      );
+
+      return res.status(200).json({
+        status: 200,
+        success: true,
+        data: updatedUserRole,
+      });
+    } else {
+      return res.status(400).json({
+        status: 400,
+        success: false,
+        message: "Please provide a valid uuid for roleId",
+      });
+    }
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while giving role a specified user",
+        error: error.message,
+      });
+    }
+  }
+};
+
+// delete a specific role
+export const deleteARole = async (req: Request, res: Response) => {
+  try {
+    if (isValidUuid(req.params.id)) {
+      const deletationRes = await deleteRoleService(req.params.id);
+
+      if (deletationRes === 0) {
+        return res.status(404).json({
+          status: 404,
+          success: false,
+          message: "Role doesn't exist!",
+        });
+      } else {
+        return res.status(200).json({
+          status: 200,
+          success: true,
+          message: "Role deleted successfully",
+          data: deletationRes,
+        });
+      }
+    } else {
+      return res.status(400).json({
+        status: 400,
+        success: false,
+        message: "Please provide a valid uuid for Role to delete",
+      });
+    }
+  } catch (error) {
+    if (error instanceof Error) {
+      return res.status(500).json({
+        status: 500,
+        success: false,
+        message: "Error occur while deleting role",
+        error: error.message,
+      });
+    }
+  }
+};
```

Comment:
@ISHIMWE-Aime I think you missed something here what if you delete a role which is already assigned to a user what happens ?

Reply:
The role will be deleted with its associations in users table and if the role is seller, the associations in product table will be deleted as well.
That all will happen. 
If the there is anything wrong please help me to understand.